datafusion/functions-aggregate/src/sum.rs

```diff
@@ -130,7 +126,42 @@ pub struct Sum {
 impl Sum {
     pub fn new() -> Self {
         Self {
-            signature: Signature::user_defined(Volatility::Immutable),
+            // Refer to https://www.postgresql.org/docs/8.2/functions-aggregate.html doc
+            // smallint, int, bigint, real, double precision, decimal, or interval.
```

Comment:
`or interval` - Does that mean that 
```
 TypeSignature::Coercible(vec![Coercion::new_exact(
                        TypeSignatureClass::Duration,
                    )]),
```
should be supported too ? Like in average.rs